src/test/java/seedu/address/model/ModelManagerTest.java

```diff
@@ -87,21 +93,69 @@ public void hasModule_moduleInModuleList_returnsTrue() {
         assertTrue(modelManager.hasModule(CS2030));
     }
 
+    @Test
+    public void deleteModule_nullModule_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> modelManager.deleteModule(null));
+    }
+
+    @Test
+    public void deleteModule_moduleNotInModuleList_throwsContactNotFoundException() {
+        // assertThrows(ContactNotFoundException.class, () -> modelManager.deleteModule(null));
+    }
+
+    @Test
+    public void deleteModule_moduleInModuleList_success() {
+        // Module module = new ModuleBuilder().build();
+        // modelManager.addModule(module);
+        // modelManager.deleteModule(module);
+        // ModelManager expectedModelManager = new ModelManager();
+        // assertEquals(modelManager, expectedModelManager);
+    }
+
+    @Test
+    public void addModule_nullModule_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> modelManager.addModule(null));
+    }
+
+    @Test
+    public void hasContact_nullContact_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> modelManager.hasContact(null));
+    }
+
+    @Test
+    public void hasContact_contactNotInContactList_returnsFalse() {
+        assertFalse(modelManager.hasContact(ALICE));
+    }
+
+    @Test
+    public void hasContact_contactInContactList_returnsTrue() {
+        modelManager.addContact(ALICE);
+        assertTrue(modelManager.hasContact(ALICE));
+    }
+
     @Test
     public void getFilteredModuleList_modifyList_throwsUnsupportedOperationException() {
         assertThrows(UnsupportedOperationException.class, () -> modelManager.getFilteredModuleList().remove(0));
     }
 
+    @Test
+    public void getFilteredContactList_modifyList_throwsUnsupportedOperationException() {
+        assertThrows(UnsupportedOperationException.class, () -> modelManager.getFilteredContactList().remove(0));
+    }
+
     @Test
     public void equals() {
         ModuleList moduleList = new ModuleListBuilder().withModule(CS2030).withModule(CS2101).build();
         ModuleList differentModuleList = new ModuleList();
+        ContactList contactList = new ContactListBuilder().withContact(ALICE).withContact(BENSON).build();
+        ContactList differentContactList = new ContactList();
+        TodoList todoList = new TodoList();
         UserPrefs userPrefs = new UserPrefs();
 
         // same values -> returns true
-        // modelManager = new ModelManager(moduleList, userPrefs);
-        // ModelManager modelManagerCopy = new ModelManager(moduleList, userPrefs);
```

Comment:
Perhaps these comments could be removed since the code here is obselete

Reply:
Yes, I will take note of these tests and modify them accordingly.